x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -167,12 +181,86 @@ func (k msgServer) ClaimMorseSupplier(
 		preClaimSupplierStake = *foundSupplier.Stake
 	}
 
+	sharedParams := k.sharedKeeper.GetParams(sdkCtx)
+	sessionEndHeight := sharedtypes.GetSessionEndHeight(&sharedParams, sdkCtx.BlockHeight())
+
+	postClaimSupplierStake := preClaimSupplierStake.Add(morseClaimableAccount.GetSupplierStake())
+	minStake := k.supplierKeeper.GetParams(ctx).MinStake
+
+	// If the claimed supplier stake is less than the minimum stake, the supplier is immediately unstaked.
+	if postClaimSupplierStake.Amount.LT(minStake.Amount) {
+		currentSessionStart := sharedtypes.GetSessionStartHeight(&sharedParams, sdkCtx.BlockHeight())
```

Comment:
This function is getting kind of massive. 

Wdyt of having a helper called "clamiSupplierAboveMinStake" and "claimSupplierBelowMinStake"?

It's godoc will also serve as a good reference for what's happening.